### PR TITLE
Implement Edit Feature

### LINKED ISSUE #90: [Refactor] [UG] [DG] Modify Edit command
## Details
- [x] Modify `edit command` to reference Trackermon's classes instead of AB3.
- [x] Add `edit command` to Trackermon's User Guide.
- [x] Add `edit use case` to Trackermon's Developer Guide.

### LINKED ISSUE #113: [Fix] Update User Guide v1.2
Fix bug/formatting issues with User Guide

### LINKED ISSUE #114: [DG] [Fix] Update Developer Guide v1.2
Fix bug/formatting issues with Developer Guide

### PR DESCRIPTION
**Pull request by ghost**
Implement Edit Feature
Update DG, UG

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -154,6 +155,24 @@ Examples:
...
+
+Format: `edit <INDEX> [n/<NAME>] [s/<STATUS>] [t/<TAG>]…​`
+* Edit the show at the specified `<INDEX>`.
+* The index refers to the index number shown in the displayed show list. (not overall)
+* The index **must be a positive integer** 1,2,3,..
+* At least one field to edit must be provided
+* Editing to an existing name is not allowed
+
+Examples:
+* `list` followed by `edit 2 n/Sailor Moo` edit 2nd show's name in Trackermon to Sailor Moon.
```
Comment (arcornior) [inline docs/UserGuide.md:171]: Should this be `edit 2 n/Sailor Moon` instead?

File: docs/UserGuide.md [hunk 2]
```diff
@@ -194,11 +213,13 @@ _Details coming soon ..._
...
-**Delete** | `delete <INDEX>`<br> e.g., `delete 3`
-**List** | `list`
-**Exit** | `exit`
+Action | Format, Examples                                                               
+--------|--------------------------------------------------------------------------------
+**Add** | `add n/<NAME> s/<STATUS> [t/<TAG>]…​` <br> e.g., `n/ReZero s/watching t/Anime` 
+**Delete** | `delete <INDEX>`<br> e.g., `delete 3`                                          
+**List** | `list`                                                                         
+**Exit** | `exit`                                                                         
+|     **Edit**    | `edit <INDEX> [n/<NAME>] [s/<STATUS>] [t/<TAG>]…​` <br> e.g., `n/ReZero s/watching t/Anime`                            |
```
Comment (arcornior) [inline docs/UserGuide.md:222]: Should there be an additional | on the left of `**Edit**`?

File: src/main/java/seedu/address/logic/commands/EditCommand.java [hunk 3]
```diff
@@ -1,227 +1,197 @@
...
+    public EditCommand(Index index, EditShowDescriptor editShowDescriptor) {
+        requireNonNull(index);
+        requireNonNull(editShowDescriptor);
+
+        this.index = index;
+        this.editShowDescriptor = new EditShowDescriptor(editShowDescriptor);
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
```
Comment (arcornior) [inline src/main/java/seedu/address/logic/commands/EditCommand.java:63]: Should this function have a JavaDoc comment?

**Review by arcornior — APPROVED** [review]
LGTM! Just a few typos to edit.

File: docs/UserGuide.md (hunk 1 above)
Comment (lowkaiwei98) [reply docs/UserGuide.md:171]: Sailor Moo is a show about magical cow saving the world

File: docs/UserGuide.md (hunk 2 above)
Comment (lowkaiwei98) [reply docs/UserGuide.md:222]: not sure why that appears but it looks fine 
image

File: src/main/java/seedu/address/logic/commands/EditCommand.java (hunk 3 above)
Comment (lowkaiwei98) [reply src/main/java/seedu/address/logic/commands/EditCommand.java:63]: cmiiw overrided methods inherit Javadocs from parent

File: docs/UserGuide.md (hunk 2 above)
Comment (Ardentsoul) [reply docs/UserGuide.md:222]: Should the `Exit` and `Edit` be swapped in the table?

Comment (lowkaiwei98) [reply docs/UserGuide.md:222]: If you want alphabetical order, i can swap

Comment (lowkaiwei98) [reply docs/UserGuide.md:222]: Personally i feel that adding latest feature to the last row would be easier unless yall got ocd then we sort by alphabet.

Comment (arcornior) [reply docs/UserGuide.md:222]: Ah I just found it weird that Edit's the only one with that `|` on the left.
Arranging by alphabetical order might help the reader when they're scanning through the commands available, but they could also `Ctrl + F`. I'd prefer alphabetical order though, just my 2 cents on this issue.

File: docs/UserGuide.md (hunk 1 above)
Comment (arcornior) [reply docs/UserGuide.md:171]: Time to change the Sailor Moon in `Trackermon to Sailor Moon` to Sailor Moo then haha.

File: docs/UserGuide.md (hunk 2 above)
Comment (lowkaiwei98) [reply docs/UserGuide.md]: Alright, i change it to alphabetical order, let me know if everything is still not to your satisfaction.

**Review by Ardentsoul — APPROVED** [review]
LGTM!

**Pull request merged**